src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp

```diff
@@ -858,19 +881,29 @@ NetworkDescription LevelZeroCompilerInDriver<TableExtension>::compile(const std:
     auto networkMeta = getNetworkMeta(graphHandle);
     networkMeta.name = model->get_friendly_name();
 
-    result = _graphDdiTableExt->pfnDestroy(graphHandle);
```

Comment:
We might still have some problems here if the compiled model is just exported in a file and do not run any inference. In this case graph should be destroyed somehow.

Reply:
Now the handle is destroyed on two path:
If CompiledModel has created executor, then executor to release handle in destructor
If CompiledModel has empty executor, then CompiledModel will call current compiler to release handle.
Since executor is member of CompiledModel, CompiledModel is member of inferRequest, the time to release handle is controlled by shared_ptr, the time to release handle is similar to old path.